cmd/thanos/receive.go

```diff
@@ -85,7 +92,7 @@ func registerReceive(app *extkingpin.App) {
 			NoLockfile:                     conf.noLockFile,
 			WALCompression:                 conf.walCompression,
 			MaxExemplars:                   conf.tsdbMaxExemplars,
-			EnableExemplarStorage:          true,
+			EnableExemplarStorage:          conf.tsdbEnableExemplarStorage,
```

Comment:
It seems like you're still missing this:
```suggestion
			EnableExemplarStorage:          conf.tsdbMaxExemplars > 0,
```
Please remove the new flag :smile: